rosbag2_transport/src/rosbag2_transport/qos.cpp

```diff
@@ -68,3 +73,126 @@ bool convert<rosbag2_transport::Rosbag2QoS>::decode(
   return true;
 }
 }  // namespace YAML
+
+namespace rosbag2_transport
+{
+Rosbag2QoS Rosbag2QoS::adapt_request_to_offers(
+  const std::string & topic_name, const std::vector<rclcpp::TopicEndpointInfo> & endpoints)
+{
+  static const Rosbag2QoS default_request{};
+  if (endpoints.empty()) {
+    return default_request;
+  }
+  size_t num_endpoints = endpoints.size();
+  size_t reliability_reliable_endpoints_count = 0;
+  size_t durability_transient_local_endpoints_count = 0;
+  for (const auto & endpoint : endpoints) {
+    const auto & profile = endpoint.qos_profile().get_rmw_qos_profile();
+    if (profile.reliability == RMW_QOS_POLICY_RELIABILITY_RELIABLE) {
+      reliability_reliable_endpoints_count++;
+    }
+    if (profile.durability == RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL) {
+      durability_transient_local_endpoints_count++;
+    }
+  }
+
+  // We set policies in order as defined in rmw_qos_profile_t
+  Rosbag2QoS request_qos = default_request;
```

Comment:
auto?

Reply:
Looking at it again, `default_request` is reused earlier where it can be returned (by value) or assigned here. I think it would make more sense to construct a new instance of `Rosbag2Qos` at either point than to define a constant global for a default instance.